go/teams/service_helper.go

```diff
@@ -1582,10 +1582,23 @@ func FindNextMerkleRootAfterRemoval(mctx libkb.MetaContext, arg keybase1.FindNex
 	}
 
 	uv := vers.ToUserVersion()
-	logPoint := team.chain().GetLastUserLogPointWithPredicate(uv, func(p keybase1.UserLogPoint) bool {
-		return p.Role == keybase1.TeamRole_NONE || p.Role == keybase1.TeamRole_READER
-	})
-	if logPoint == nil {
+	logPoints := team.chain().inner.UserLog[uv]
+	demotionPredicate := func(p keybase1.UserLogPoint) bool {
+		if arg.AnyRoleAllowed {
+			return !p.Role.IsReaderOrAbove()
+		}
+		return !p.Role.IsWriterOrAbove()
+	}
+	var earliestDemotion int
+	var logPoint keybase1.UserLogPoint
+	for i := len(logPoints) - 1; i >= 0; i-- {
+		if demotionPredicate(logPoints[i]) {
+			earliestDemotion = i
+		} else if earliestDemotion != 0 {
+			logPoint = logPoints[earliestDemotion].DeepCopy()
```

Comment:
hmm, wait a second... What if AnyRoleAllowed is false and the sequence is:

Link 1: Writer
Link 2: Reader
Link 3: Writer
Link 4: Reader

Won't this return Link 2?  It should return link 4.

Reply:
oh blerg. you're right. i'm not exiting out of the loop once the `logPoint` is set. :/

Reply:
updated the test and added the `break`